pkg/raft/tracker/progress.go

```diff
@@ -433,3 +433,13 @@ func (m ProgressMap) String() string {
 	}
 	return buf.String()
 }
+
+// BasicProgress contains a subset of fields from Progress.
```

Comment:
After we bump to Go 1.23, we will be able to implement an iterator into the `Progress` map, without needed these various "basic/non-basic" introspection options.

While here, can you add a TODO to revisit/remove these helpers, with Go 1.23?

Reply:
https://tip.golang.org/doc/go1.23#language